### ISSUE TITLE
@Nullable List is null even when json field is an empty array

### ISSUE THREAD

**Issue by alexio** [immutables/immutables#355]
Found issue when upgrading from `2.1.10` to `2.2.2` when one of my tests failed. By marking a List field `@Nullable`, the output object from gson will be null when the json field is an empty array. I assume this is a bug and not intended.

Made a sample project with a test to show the issue: https://github.com/alexio/ImmutablesBug-NullList/blob/master/README.md

**Comment by elucash** [comment]
Indeed, this should deserialize as an empty list, not as null, just hope it would not hard to fix. Thank you for the bug report!

**Comment by alexio** [comment]
Awesome! Thanks! :D
